### PR TITLE
Add ping call

### PR DESCRIPTION
**Pull request by hagail**

### CODE REVIEW

File: lib/ringcaptcha/api.rb [hunk 1]
```diff
@@ -35,5 +35,16 @@ def self.do_http_request(http, request)
         end
       end
     end
+
```
Comment (TalyaS) [inline lib/ringcaptcha/api.rb:38]: LGTM

File: lib/ringcaptcha.rb [hunk 2]
```diff
@@ -33,6 +33,12 @@ def verify(app_key, token, code)
       return api(app_key).call(@api_key, app_key, 'verify', token: token, code: code)
     end
 
+    def ping
+      [Ringcaptcha::API.service_up?, nil]
+    rescue StandardError => e
```
Comment (TalyaS) [inline lib/ringcaptcha.rb:38]: why do you rescue standard errors only?

**Comment by tomerlevi444** [comment]
Approved

**Comment by moriaam** [comment]
Approved

File: lib/ringcaptcha/api.rb [hunk 3]
```diff
@@ -35,5 +35,16 @@ def self.do_http_request(http, request)
...
+    def self.service_up?
+      uri = URI.parse("https://api.ringcaptcha.com")
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+      response = http.request(Net::HTTP::Post.new("/status"))
+
+      response.is_a?(Net::HTTPSuccess) && response.body == "WE ARE ALIVE ;-)"
```
Comment (jondot) [inline lib/ringcaptcha/api.rb:47]: Any reason for POST over GET?
Did you consider not checking for the response body? i.e. just 200 OK would be enough.
You could then also avoid pulling the body and use just a more efficient HEAD request.

**Comment by danevron** [comment]
Approved

File: lib/ringcaptcha.rb (hunk 2 above)
Comment (cdambo) [inline lib/ringcaptcha.rb:38]: Exception?

**Comment by cdambo** [comment]
Approved

File: lib/ringcaptcha.rb (hunk 2 above)
Comment (hagail) [inline lib/ringcaptcha.rb:38]: We had a similar discussion in another PR

File: lib/ringcaptcha/api.rb (hunk 3 above)
Comment (hagail) [inline lib/ringcaptcha/api.rb:47]: You're right, I should used GET. I'll change that.
Why not check the response body? They could for instance return 200 with a body of "we're down". I prefer to be more strict.

**Pull request merged**